cvinfer/io.py

```diff
@@ -202,9 +204,14 @@ def _next(self):
             # require looping because of frame skipping feature
             while True:
                 is_valid, frame = self._video_handle.read()
+                if not self._first_frame:
+                    self.frame_count += 1
+                else:
+                    self._first_frame = False
+
                 if is_valid:
                     # if read successfully
-                    if self._skip_frame_frequency == self._skip_count:
+                    if self.frame_count % self._skip_frame_frequency == 0:
```

Comment:
I think the meaning of `skip_frame_frequency` is slightly different than what I had originally. 
If `skip_frame_frequency=2`, your implementation would be taking frame at frame indices 0, 2, 4, 6, ... right? meaning skipping only 1 frame right? 
I was thinking like if `skip_frame_frequency=2` then we should read index 0, skip 2 frames, read index 3, skip 2 frames, read index 6, and so on. The term frequency here probably sounds misleading in that sense and `skip_frame_interval` might be a better term. 

But how does your implementation work when `skip_frame_frequency=0` (default value, no skipping) in the denominator?